app/controllers/users_controller.rb

```diff
@@ -21,9 +21,34 @@ def create
       redirect_to register_path
     end
   end
-    
+
   def discover
-    
+    @user = User.find(params[:user_id])
+    @top_rated_movies = MoviesService.new.top_rated
+  end
+
+  def movies
+    @user = User.find(params[:user_id])
+    query = params[:q]
+    movies_data = []
+
+    @title = if query == 'top_rated'
+               'Top Rated Movies'
+             elsif query.present?
+               "Search Results for '#{query}'"
+             else
+               'Error: No Query'
+             end
+
+    movies_data = if query == 'top_rated'
+                    MoviesService.new.top_rated.first(20)
+                  elsif query.present?
+                    MoviesService.new.search(query).first(20)
+                  else
+                    []
+                  end
+
+    @movies = movies_data.map { |movie_data| Movie.new(movie_data) }
```

Comment:
Looks good! We'll get this sorted into another controller/facade/etc on a refactoring branch